linkis-computation-governance/linkis-manager/linkis-application-manager/src/main/scala/org/apache/linkis/manager/am/service/engine/DefaultEngineCreateService.scala

```diff
@@ -298,9 +297,9 @@ class DefaultEngineCreateService
       resource,
       timeout
     ) match {
-      case AvailableResource(ticketId) =>
```

Comment:
Why do we need to return Json here?

Reply:
seems it is used for getting resource [resourceTicketId, resource] info if we got enough resource, or we will throw not enough resource.
```
    val emNode = choseNode.get.asInstanceOf[EMNode]
    // 4. 请求资源
    val (resourceTicketId, resource) =
      requestResource(engineCreateRequest, labelFilter.choseEngineLabel(labelList), emNode, timeout)

    // 5. 封装engineBuildRequest对象,并发送给EM进行执行
    val engineBuildRequest = EngineConnBuildRequestImpl(
      resourceTicketId,
      labelFilter.choseEngineLabel(labelList),
      resource,
      EngineConnCreationDescImpl(
        engineCreateRequest.getCreateService,
        engineCreateRequest.getDescription,
        engineCreateRequest.getProperties
      )
    )
```